### PR TITLE
[Indexer] Refactor and Simplify indexer, Genesis tx update indexer

### PR DESCRIPTION
**Pull request by baichuan3**
## Summary

1. Refactor and Simplify indexer, remove action_raw and transaction_raw from indexer
2. Save the genesis txs to indexer, including states, transaction and event indexer
3. The indexer object_type include full StructTag 
4. Fixup Indexer object state not consistent with state bug.

- Closes 
- #1737 
- #1753 
- #1759

### CODE REVIEW

**Comment by jolestar** [comment]
The first bug of #1759 should be fixed in this PR. @pause125

File: crates/rooch-indexer/src/models/transactions.rs [hunk 1]
```diff
@@ -82,22 +69,33 @@ impl From<IndexedTransaction> for StoredTransaction {
...
+        let tx_accumulator_root = H256::from_str(transaction.tx_accumulator_root.as_str())?;
+        let state_root = H256::from_str(transaction.state_root.as_str())?;
+        let event_root = H256::from_str(transaction.event_root.as_str())?;
+
+        let indexer_transaction = IndexerTransaction {
+            tx_hash,
+            tx_order: transaction.tx_order as u64,
+            sequence_number: transaction.sequence_number as u64,
+            sender,
+            action: None,
```
Comment (jolestar) [inline crates/rooch-indexer/src/models/transactions.rs:87]: Should we remove the action field from IndexerTransaction? If we do not store it.

File: crates/rooch-indexer/src/models/states.rs [hunk 2]
```diff
@@ -69,16 +67,15 @@ impl StoredObjectState {
     pub fn try_into_indexer_global_state(&self) -> Result<IndexerObjectState, anyhow::Error> {
         let object_id = ObjectID::from_str(self.object_id.as_str())?;
         let owner = RoochAddress::from_str(self.owner.as_str())?;
-
         let object_type = StructTag::from_str(self.object_type.as_str())?;
-        let state_root = H256::from_str(self.state_root.as_str())?;
+        let state_root = RoochAddress::from_str(self.state_root.as_str())?;
```
Comment (jolestar) [inline crates/rooch-indexer/src/models/states.rs:71]: Should use H256 here?

File: crates/rooch-indexer/src/proxy/mod.rs [hunk 3]
```diff
@@ -36,6 +36,27 @@ impl IndexerProxy {
...
+    pub async fn update_indexer(
+        &self,
+        root: RootObjectEntity,
+        ledger_transaction: LedgerTransaction,
+        execution_info: TransactionExecutionInfo,
+        moveos_tx: VerifiedMoveOSTransaction,
+        events: Vec<Event>,
+        state_change_set: StateChangeSet,
+    ) -> Result<()> {
+        self.actor
```
Comment (jolestar) [inline crates/rooch-indexer/src/proxy/mod.rs:48]: We can use the `actor::notify` function in the future and remove the `tokio::spaw.`

File: crates/testsuite/features/cmd.feature [hunk 4]
```diff
@@ -114,22 +114,22 @@ Feature: Rooch CLI integration tests
...
     Then assert: "{{$.rpc[-1].data[0].indexer_event_id.tx_order}} == 1"
-    Then assert: "{{$.rpc[-1].next_cursor.tx_order}} == 1"
+    Then assert: "{{$.rpc[-1].next_cursor.tx_order}} == 0"
     Then assert: "{{$.rpc[-1].has_next_page}} == false"
 
     # Sync states
     Then cmd: "rpc request --method rooch_queryObjectStates --params '[{"object_type":"0x3::coin::CoinInfo"}, null, "10", {"descending": true,"showDisplay":false}]'"
-    #Then assert: "{{$.rpc[-1].data[0].tx_order}} == 0"
+    Then assert: "{{$.rpc[-1].data[0].tx_order}} == 1"
     Then assert: "{{$.rpc[-1].data[0].object_type}} == 0x3::coin::CoinInfo"
```
Comment (jolestar) [inline crates/testsuite/features/cmd.feature:133]: The `object_type` should be `0x3::coin::CoinInfo<0x3::gas_coin::GasCoin>`

**Comment by baichuan3** [comment]
> The first bug of #1759 should be fixed in this PR. @pause125

already fixup

**Review by jolestar — APPROVED** [review]

**Pull request merged**